packages/nextjs/components/blockexplorer/TransactionsTable.tsx

```diff
@@ -0,0 +1,139 @@
+import Link from "next/link";
+import { TransactionResponse } from "@ethersproject/abstract-provider";
+import { ethers } from "ethers";
+import { Address } from "~~/components/scaffold-eth";
+
+type TransactionWithFunction = TransactionResponse & { functionName?: string };
+
+// Define your types
+interface Block {
+  hash: string;
+  transactions: TransactionWithFunction[];
+  timestamp: number;
+  number: number;
+}
+
+interface ConfiguredNetwork {
+  nativeCurrency: {
+    symbol: string;
+  };
+}
+
+interface TransactionReceipts {
```

Comment:
Since in `pages/blockexplorer.ts` we have defined state type as : 

```ts
interface TransactionReceipts {
  [key: string]: TransactionReceipt;
}
```